modules/reactor/doc/content/source/meshgenerators/ReactorMeshParams.md

```diff
@@ -0,0 +1,17 @@
+# ReactorMeshParams
+
+!syntax description /Mesh/ReactorMeshParams
+
+## Overview
+
+The `ReactorMeshParams` object stores persistent mesh information about a reactor's geometry for use with [PinMeshGenerator](/PinMeshGenerator.md), [AssemblyMeshGenerator](/AssemblyMeshGenerator.md), and [CoreMeshGenerator](/CoreMeshGenerator.md). This is where the geometry type ([!param](/Mesh/ReactorMeshParams/geom) as 'Square' or 'Hex' for cartesian and hexagonal definitions respectively) and the number of dimensions of the mesh ([!param](/Mesh/ReactorMeshParams/dim) 2 or 3D) is declared and persistently enforced for the rest of the mesh definition. If the mesh is to be 3-dimensional, this is also where the axial information is declared ([!param](/Mesh/ReactorMeshParams/axial_regions) and [!param](/Mesh/ReactorMeshParams/axial_mesh_intervals)).
+
```

Comment:
I think this PR needs a convention to set reporting IDs and/or block IDs.

Possibly something like: ID = assembly_number * A * B + pin_number * B + region number
B = number of regions in a pin
A = number of regions in an assembly.

Then for background, ducts, top/bottom:
ID = offset + same_formula
offset is big enough to avoid collisions

This would be an alternative (both would be offered) to setting IDs manually.
Right now we have a mix of setting IDs manually and basic procedures (20000 + assembly_number+ 1)
@YaqiWang any opinion on this?

Reply:
Is there an actual use case for automatically setting IDs? The reason why we removed the `procedural_ids` option and left it up to the users to define the block ID's throughout the reactor mesh is that we are assuming the users have some idea of how the material-region mapping is going to be before the mesh generation stage, so they can avoid having to remap all block ID's to match material ID's later on in the input generation stage and take care of it at this stage. The difficulty with implementing a mapping like you propose is that in its current form pins / assemblies do not pass information about how many regions they have to neighboring pins / assemblies so it won't be straightforward to determine what the starting ID should be for a neighboring pin / assembly given the block ID assignment of the current assembly / pin.

Reply:
@olinwc